spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/observation/ObservedAppDeployer.java

```diff
@@ -0,0 +1,232 @@
+		context.setAppId(id);
+		Observation observation = AppDeployerDocumentedObservation.UNDEPLOY_OBSERVATION.start(this.customConvention, DEFAULT_CONVENTION, context, this.observationRegistry);
+		observation.scoped(() -> {
+			observation.event(AppDeployerDocumentedObservation.Events.DEPLOYER_START);
+			this.delegate.undeploy(id);
+			registerListener(observation, id);
+			return id;
+		});
+	}
+
+	@Override
+	public AppStatus status(String id) {
+		AppDeployerContext context = new AppDeployerContext(environmentInfo());
+		context.setAppId(id);
+		Observation observation = AppDeployerDocumentedObservation.STATUS_OBSERVATION.start(this.customConvention, DEFAULT_CONVENTION, context, this.observationRegistry);
+		return observation.observe(() -> this.delegate.status(id));
+	}
+
+	@Override
+	public Mono<AppStatus> statusReactive(String id) {
+		AppDeployerContext context = new AppDeployerContext(environmentInfo());
+		context.setAppId(id);
+		return Mono.defer(() -> Mono.just(AppDeployerDocumentedObservation.STATUS_OBSERVATION.start(this.customConvention, DEFAULT_CONVENTION, context, this.observationRegistry)))
+				.flatMap(observation -> this.delegate.statusReactive(id).doOnNext(appStatus -> observation.stop()));
+	}
+
+	@Override
+	public Flux<AppStatus> statusesReactive(String... ids) {
+		AppDeployerContext context = new AppDeployerContext(environmentInfo());
+		context.setAppId(String.join(",", ids));
+		return Flux.defer(() -> Flux.just(AppDeployerDocumentedObservation.STATUS_OBSERVATION.start(this.customConvention, DEFAULT_CONVENTION, context, this.observationRegistry)))
+				.flatMap(observation -> this.delegate.statusesReactive(ids).doOnNext(appStatus -> observation.stop()));
+	}
+
+	@Override
+	public RuntimeEnvironmentInfo environmentInfo() {
+		return this.delegate.environmentInfo();
+	}
+
+	@Override
+	public String getLog(String id) {
+		AppDeployerContext context = new AppDeployerContext(environmentInfo());
+		context.setAppId(id);
+		Observation observation = AppDeployerDocumentedObservation.GET_LOG_OBSERVATION.start(this.customConvention, DEFAULT_CONVENTION, context, this.observationRegistry);
+		return observation.observe(() -> this.delegate.getLog(id));
+	}
+
+	@Override
+	public void scale(AppScaleRequest appScaleRequest) {
+		AppDeployerContext context = new AppDeployerContext(environmentInfo());
+		context.setScaleRequest(appScaleRequest);
+		Observation observation = AppDeployerDocumentedObservation.SCALE_OBSERVATION.start(this.customConvention, DEFAULT_CONVENTION, context, this.observationRegistry);
+		observation.observe(() -> this.delegate.scale(appScaleRequest));
+	}
+
+	private static final class PreviousAndCurrentStatus {
+
+		private final Observation observation;
+
+		private AppStatus current;
+
+		private AppStatus previous;
+
+		private PreviousAndCurrentStatus(Observation observation) {
+			this.observation = observation;
+			if (log.isDebugEnabled()) {
```

Comment:
Thank you for adding `LogAccessor`. Do you mind updating these calls to take advantage of it? These become 
```java
log.debug(() -> "Current observation is [" + observation + "]";
```
We get to drop the pesky "is<level>Enabled" as the internal impl does that and only calls supplier when its enabled.